model/riscv_insts_zbs.sail

```diff
@@ -0,0 +1,94 @@
+/* ****************************************************************** */
+/* This file specifies the instructions in the BitManip Zbs extension */
+
+
+/* ****************************************************************** */
+union clause ast = ZBS_IOP : (bits(6), regidx, regidx, biop)
+
+mapping clause encdec = ZBS_IOP(shamt, rs1, rd, RISCV_BCLRI) if haveZbs() & (sizeof(xlen) == 64 | shamt[5] == bitzero)
+  <-> 0b010010 @ shamt @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec = ZBS_IOP(shamt, rs1, rd, RISCV_BEXTI) if haveZbs() & (sizeof(xlen) == 64 | shamt[5] == bitzero)
+  <-> 0b010010 @ shamt @ rs1 @ 0b101 @ rd @ 0b0010011
+
+mapping clause encdec = ZBS_IOP(shamt, rs1, rd, RISCV_BINVI) if haveZbs() & (sizeof(xlen) == 64 | shamt[5] == bitzero)
+  <-> 0b011010 @ shamt @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec = ZBS_IOP(shamt, rs1, rd, RISCV_BSETI) if haveZbs() & (sizeof(xlen) == 64 | shamt[5] == bitzero)
+  <-> 0b001010 @ shamt @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping zbs_iop_mnemonic : biop <-> string = {
+  RISCV_BCLRI <-> "bclri",
+  RISCV_BEXTI <-> "bexti",
+  RISCV_BINVI <-> "binvi",
+  RISCV_BSETI <-> "bseti"
+}
+
+mapping clause assembly = ZBS_IOP(shamt, rs1, rd, op)
+  <-> zbs_iop_mnemonic(op) ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ hex_bits_6(shamt)
+
+function clause execute (ZBS_IOP(shamt, rs1, rd, op)) = {
+  let rs1_val = X(rs1);
+  let result : xlenbits = match op {
+    RISCV_BCLRI => if sizeof(xlen) == 32
+                     then rs1_val & ( ~ (EXTZ(0b1) << shamt[4..0]))
+                     else rs1_val & ( ~ (EXTZ(0b1) << shamt[5..0])),
+    RISCV_BEXTI => if sizeof(xlen) == 32
+                     then (rs1_val >> shamt[4..0]) & EXTZ(0x1)
+                     else (rs1_val >> shamt[5..0]) & EXTZ(0x1),
+    RISCV_BINVI => if sizeof(xlen) == 32
+                     then rs1_val ^ (EXTZ(0b1) << shamt[4..0])
+                     else rs1_val ^ (EXTZ(0b1) << shamt[5..0]),
+    RISCV_BSETI => if sizeof(xlen) == 32
+                     then rs1_val | (EXTZ(0b1) << shamt[4..0])
+                     else rs1_val | (EXTZ(0b1) << shamt[5..0])
+  };
```

Comment:
```suggestion
  let rs1_val = X(rs1);
  let mask : xlenbits = if sizeof(xlen) == 32
                        then EXTZ(0b1) << shamt[4..0]
                        else EXTZ(0b1) << shamt;
  let result : xlenbits = match op {
    RISCV_BCLRI => rs1_val & ~(mask),
    RISCV_BEXTI => EXTZ(if sizeof(xlen) == 32
                        then rs1_val[shamt[4..0]]
                        else rs1_val[shamt[5..0]]),
    RISCV_BINVI => rs1_val ^ mask,
    RISCV_BSETI => rs1_val | mask
  };
```

or

```suggestion
  let rs1_val = X(rs1);
  let mask : xlenbits = if sizeof(xlen) == 32
                        then EXTZ(0b1) << shamt[4..0]
                        else EXTZ(0b1) << shamt;
  let result : xlenbits = match op {
    RISCV_BCLRI => rs1_val & ~(mask),
    RISCV_BEXTI => bool_to_bits((rs1_val & mask) != zeros()),
    RISCV_BINVI => rs1_val ^ mask,
    RISCV_BSETI => rs1_val | mask
  };
```

not sure which is better, EXTI doesn't quite fit in with the rest, personally lean towards the latter to make it fit better even though the former is more idiomatic viewed on its own

Reply:
Done.